overlord/hookstate/hooks.go

```diff
@@ -123,7 +129,12 @@ func (h *gateAutoRefreshHookHandler) Done() (err error) {
 	ctx.Lock()
 	defer ctx.Unlock()
 
-	snapName := h.context.InstanceName()
+	snapName := ctx.InstanceName()
+	snapInfo, err := snapstate.CurrentInfo(st, snapName)
+	if err != nil {
+		return err
+	}
+	snapRev := snapInfo.SnapRevision()
```

Comment:
we could also use IsLocked below to find and reuse the current value instead?

Reply:
great point, updated, thank you!